src/cdn_definitions/schema.yaml

```diff
@@ -11,7 +11,211 @@ $id: http://release-engineering.github.io/cdn-definitions/schema.json
 
 
 definitions:
+  arch:
+    type: string
+    pattern: "^(x86|x86\\_64|ia64|i386|ppc|ppc64|ppc64le|s390|s390x|aarch64)$"
+
+  arch_list:
+    type: array
+    items:
+      $ref: "#/definitions/arch"
+
+  release_version:
+    type: string
+    pattern: "^[0-9]+\\.[0-9]+$"
+
+  major_version:
+    type: string
+    pattern: "^[0-9]+$"
+
+  signing_key:
+    type: string
+    pattern: "^(37036783|A5787476|FD431D51|F21541EB|37017186|897DA07A|DB42A60E|\\[\\])$"
+
+  product_id:
+    type: string
+    pattern: "^[0-9]+$"
+
+  product_id_list:
+    type: array
+    items:
+      $ref: "#/definitions/product_id"
+
+  signing_key_list:
+    type: array
+    items:
+      $ref: "#/definitions/signing_key"
+
+  release_version_list:
+    type: array
+    items:
+      $ref: "#/definitions/release_version"
+
+  major_version_list:
+    type: array
+    items:
+      $ref: "#/definitions/major_version"
+
+  major_or_release_version:
+    type: string
+    pattern: "^[0-9]+(\\.[0-9]+)*$"
+
+  # various product definition mappings
+  arch_to_version_list_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/arch"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/major_version_list"
+    additionalProperties: false
+
+  major_to_release_version_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/major_version"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/release_version"
+    additionalProperties: false
+
+  release_stream_mapping:
+    type: object
+    properties:
+      aus:
```

Comment:
I think allowing arbitrary streams here might be a good idea, they do change sometimes and it might be desired to prepare for introducing a new stream before it's announced publicly.